File: backend/src/main/java/project/backend/global/exception/ErrorResponse.java
package project.backend.global.exception;

import java.util.Map;
import lombok.Builder;
import lombok.Getter;
import org.springframework.validation.FieldError;
import project.backend.global.exception.errorcode.ErrorCode;

@Builder
@Getter
public class ErrorResponse {

	private String code;
	private String message;
	private Map<String, Object> details;

	public static ErrorResponse toResponse(ErrorCode errorCode) {
		return ErrorResponse.builder()
			.code(errorCode.code)
			.message(errorCode.message)
			.build();
	}

	public static ErrorResponse toResponse(FieldError error) {
		return ErrorResponse.builder()
			.code("VALIDATION_FAILED")
			.message(error.getDefaultMessage())
			.build();
	}

}

File: backend/src/main/java/project/backend/global/exception/ErrorResponse.kt
package project.backend.global.exception

import org.springframework.validation.FieldError
import project.backend.global.exception.errorcode.ErrorCode

data class ErrorResponse(
    val code: String,
    val message: String?
) {
    companion object {
        fun toResponse(errorCode: ErrorCode): ErrorResponse =
            ErrorResponse(errorCode.code, errorCode.message)

        fun toResponse(error: FieldError): ErrorResponse =
            ErrorResponse("VALIDATION_FAILED", error.defaultMessage)
    }
}
